### ISSUE TITLE
USE_DATA_CLASS: False positive for Annotations

### ISSUE THREAD

**Issue by leinardi** [saveourtool/diktat#1469]
I have this annotation:
```kotlin
@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.CLASS)
annotation class NavGraphDestination(
    val name: String = Defaults.NULL,
    val routePrefix: String = Defaults.NULL,
    val deepLink: Boolean = false,
) {
    object Defaults {
        const val NULL = "@null"
    }
}
```
And Diktat complains that
```
There are 1 unfixed errors:
Error on line: 3, column: 1 cannot be fixed automatically
[USE_DATA_CLASS] this class can be converted to a data class: NavGraphDestination
java.lang.AssertionError: There are 1 unfixed errors:
```

I guess that the `DataClassesRule` should check if a class is an `annotation`, beside checking for `data class` and `interface`:

https://github.com/saveourtool/diktat/blob/54c8ccefc3009b8dd2a493c30590168318765b89/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/DataClassesRule.kt#L46

I'm having this with Diktat 1.2.1. Let me know if you need more info.

**Comment by akuleshov7** [comment]
Hey, yeah, nice catch, thank you!
We completely forgot about that 🤦 

We are already fixing this rule in https://github.com/saveourtool/diktat/issues/1444 and we will fix this one also in 1.2.2 release.

Thank you very much for this report :)

**Comment by akuleshov7** [comment]
By for now you can easily `@Suppress` this inspection

**Comment by akuleshov7** [comment]
fixed in #1471
